### ISSUE TITLE
Not compiling

### ISSUE THREAD

**Issue by mikob** [ltratt/hk#2]
I like the simplicity of this script but I couldn't get this to install on Linux Mint 21.2 which uses X11. I have libx11-dev installed. Also tried added -L/usr/lib... Seems like it should be something I'm missing here?
```
gcc -Wl,-z,relro,-z,now -lX11 -L/usr/X11/lib -L/usr/lib -o hk hk.o
/usr/bin/ld: hk.o: in function `ignorable_modifiers':
/home/mikob/tools/hk/hk.c:49: undefined reference to `XKeysymToKeycode'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:52: undefined reference to `XGetModifierMapping'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:61: undefined reference to `XFreeModifiermap'
/usr/bin/ld: hk.o: in function `parse_modifier':
/home/mikob/tools/hk/hk.c:100: undefined reference to `XStringToKeysym'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:103: undefined reference to `XKeysymToKeycode'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:104: undefined reference to `XGetModifierMapping'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:115: undefined reference to `XFreeModifiermap'
/usr/bin/ld: hk.o: in function `parse_key':
/home/mikob/tools/hk/hk.c:126: undefined reference to `XStringToKeysym'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:130: undefined reference to `XKeysymToKeycode'
/usr/bin/ld: hk.o: in function `main':
/home/mikob/tools/hk/hk.c:199: undefined reference to `XOpenDisplay'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:208: undefined reference to `XGrabKey'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:211: undefined reference to `XGrabKey'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:215: undefined reference to `XSelectInput'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:218: undefined reference to `XNextEvent'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:277: undefined reference to `XCloseDisplay'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:229: undefined reference to `XQueryKeymap'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:244: undefined reference to `XkbKeycodeToKeysym'
/usr/bin/ld: /home/mikob/tools/hk/hk.c:255: undefined reference to `XKeysymToString'
collect2: error: ld returned 1 exit status
make: *** [Makefile:17: hk] Error 1
```

**Comment by ltratt** [comment]
I don't know mint, but I can tell you my very stupid "trick" that *sometimes* works to solve this. I basically do something like `grep -lr XKeysymToKeycode /` and see what `.so` files it outputs. If it doesn't output anything, that suggests I'm missing a library. But sometimes I find a library is installed in a surprising place and then I pass it to `-L` in the compile line and see if it works. If you do find where this library is located, please let me know, as I might want to update the `configure` script to search that directory!

**Comment by ltratt** [comment]
I managed to find a Linux box that replicates this problem. https://github.com/ltratt/hk/commit/c0f39fc8ca3039d08b51df18771227ea7c62cb6c should fix this. Please let me know if it does or not -- thanks!

**Comment by ltratt** [comment]
Fixed in hk-0.3.2 https://tratt.net/laurie/src/hk/.

**Comment by mikob** [comment]
Cool, compiles now, thanks!
